src/bionics.cpp

```diff
@@ -996,35 +993,37 @@ bool player::install_bionics( const itype &type, int skill_level, bool autodoc )
     }
 
     if( chance_of_success >= 100 ) {
-        if( !query_yn(
+        if( !g->u.query_yn(
                 _( "Are you sure you wish to install the selected bionic?" ),
                 100 - chance_of_success ) ) {
             return false;
         }
     } else {
-        if( !query_yn(
+        if( !g->u.query_yn(
                 _( "WARNING: %i percent chance of genetic damage, blood loss, or damage to existing bionics! Continue anyway?" ),
                 ( 100 - int( chance_of_success ) ) ) ) {
             return false;
         }
     }
 
     // Practice skills only if conducting manual self-installation
-    if( ( skill_level == -1 ) && !autodoc ) {
-        practice( skilll_electronics, int( ( 100 - chance_of_success ) * 1.5 ) );
-        practice( skilll_firstaid, int( ( 100 - chance_of_success ) * 1.0 ) );
-        practice( skilll_mechanics, int( ( 100 - chance_of_success ) * 0.5 ) );
+    if( !autodoc ) {
+        installer.practice( skilll_electronics, int( ( 100 - chance_of_success ) * 1.5 ) );
+        installer.practice( skilll_firstaid, int( ( 100 - chance_of_success ) * 1.0 ) );
+        installer.practice( skilll_mechanics, int( ( 100 - chance_of_success ) * 0.5 ) );
     }
 
     int success = chance_of_success - rng( 0, 99 );
     if( success > 0 ) {
-        add_memorial_log( pgettext( "memorial_male", "Installed bionic: %s." ),
-                          pgettext( "memorial_female", "Installed bionic: %s." ),
-                          bioid->name.c_str() );
-
+        if( is_player() ) {
+            add_memorial_log( pgettext( "memorial_male", "Installed bionic: %s." ),
+                              pgettext( "memorial_female", "Installed bionic: %s." ),
+                              bioid->name.c_str() );
+        }
         if( bioid->upgraded_bionic ) {
             remove_bionic( bioid->upgraded_bionic );
-            //~ %1$s - name of the bionic to be upgraded (inferior), %2$s - name of the upgraded bionic (superior).
+            //~ %1$s - name of the bionic to be upgraded (inferior),
+            //  %2$s - name of the upgraded bionic (superior).
```

Comment:
Are you sure translation comments shouldn't be on one line? Here and below.

Reply:
Fixed.